grafana/grafana
# Elasticsearch: Run getTerms trough resource call if enableElasticsearchBackendQuerying enabled

In this PR, we are use ES backend's `CallResource`  to run `getTerms` trough resource call if `enableElasticsearchBackendQuerying` is enabled.

Part of #54011

I have decided to create wrapper methods for `getResource` and `postResource`. This is because for post requests we want to update header to `'content-type' = 'application/x-ndjson'` and it felt cohesive to then have wrapper methods for both. 

How to test this:
1. Run `make devenv sources=elastic`
1. Run `cd devenev` and then `./setup.sh` to update `Datasource tests - Elasticsearch complex` dashboard to include template variables
1. Make sure that `enableElasticsearchBackendQuerying` is enabled
1. Open `Datasource tests - Elasticsearch complex` dashboard
1. Ensure that all template variables work and have values
1. Select some variables and go to last row with template variables - make sure that graph look as expected
